src/otx/core/data/transform_libs/mmdet_inst_seg.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Helper to support MMDET Instance Segmentation data transform functions."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable
+
+import numpy as np
+import torch
+from datumaro import Polygon
+from mmdet.datasets.transforms import (
+    LoadAnnotations as MMDetLoadAnnotations,
+)
+from mmdet.datasets.transforms import (
+    PackDetInputs as MMDetPackDetInputs,
+)
+from mmdet.registry import TRANSFORMS
+from mmdet.structures.mask import BitmapMasks, PolygonMasks
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.instance_segmentation import InstanceSegDataEntity
+
+from .mmcv import MMCVTransformLib
+
+if TYPE_CHECKING:
+    from mmengine.registry import Registry
+
+    from otx.core.config.data import SubsetConfig
+
+
+@TRANSFORMS.register_module(force=True)
```

Comment:
This registration can affect to the detection task's behavior, which uses the `mmdet` data pipeline. I think that you need to unify this with `src/otx/core/data/transform_libs/mmdet.py`.

Reply:
Could you provide more details on how the registration of `LoadAnnotations` might impact the detection task? As per the current design, Instance Segmentation has introduced a distinct `MMDetInstSegTransformLib` separate from `detection.MMDetTransformLib`, indicating a deliberate divergence, and should work separately.

My concern is that unifying these processes might result in the nesting of `LoadAnnotation` and `PackDetInputs` with two different data entity checks (InstanceSegDataEntity and DetDataEntity). This distinction is crucial as detection tasks don't need masks, while instance segmentation tasks do. If we proceed with unification, PackDetInputs would then need to return two different data entities.

Reply:
> Could you provide more details on how the registration of LoadAnnotations might impact the detection task? As per the current design, Instance Segmentation has introduced a distinct MMDetInstSegTransformLib separate from detection.MMDetTransformLib, indicating a deliberate divergence, and should work separately.

Let assume that a user uses OTX as Python API, not CLI. Than, the order of importing `src/otx/core/data/transform_libs/mmdet.py` and `src/otx/core/data/transform_libs/mmdet_inst_seg.py` can affect to their runtime behavior. This is hidden and hard to find.

> My concern is that unifying these processes might result in the nesting of LoadAnnotation and PackDetInputs with two different data entity checks (InstanceSegDataEntity and DetDataEntity). This distinction is crucial as detection tasks don't need masks, while instance segmentation tasks do. If we proceed with unification, PackDetInputs would then need to return two different data entities.

I think we put our data entity in `results["__otx__"]` at the very first pipeline (`LoadImageFromFile`). We can make branches by investigating the type of `results["__otx__"]`.